connect-location/src/main/java/com/ifttt/location/ConnectionRefresher.java

```diff
@@ -0,0 +1,44 @@
+package com.ifttt.location;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+import androidx.work.Worker;
+import androidx.work.WorkerParameters;
+import com.ifttt.connect.Connection;
+import com.ifttt.connect.ConnectionApiClient;
+import com.ifttt.connect.ErrorResponse;
+import com.ifttt.connect.api.PendingResult;
+
+public final class ConnectionRefresher extends Worker {
+
+    private final ConnectionApiClient connectionApiClient;
+
+    public ConnectionRefresher(Context context, WorkerParameters params) {
+        super(context, params);
+        connectionApiClient = ConnectLocation.getInstance().connectionApiClient;
+    }
+
+    @Override
+    @NonNull
+    public Result doWork() {
+        String connectionId = getInputData().getString("connectionId");
```

Comment:
I think we should move the CredentialProvider method call here:
* so that we don't have to expose UserTokenAsyncTask
* so that we can fail and retry on this worker if for whatever reason CredentialProvider fails (with try-catch).

Reply:
If we cannot get user token, we should not proceed, because without it we will not get any potential UserFeature.

Reply:
This is part of the periodic work request. We probably don't need to fetch the user token each time..

Reply:
A user token could be refreshed when connection is enabled, that can be done in the `onStateChanged` method. I added a TODO comment there.

Reply:
> This is part of the periodic work request. We probably don't need to fetch the user token each time..

Right, we can check if the ApiClient has a token already, and only use CredentialProvider if not. This is also used to account for cases where for whatever reason the ApiClient doesn't have user token before we make the API call. I don't think we can assume that the user token is correctly set in the ApiClient. 

> A user token could be refreshed when connection is enabled, that can be done in the onStateChanged method. I added a TODO comment there.

Not sure if I understand, why is the user token refresh related to this?